### PR TITLE
Feature/list technologies

### PR DESCRIPTION
**Pull request by felipegc**
There are some changes made by Felipe Roris and Felipe Gonçalves de Castro. This Branch need to be merged because if somebody starts a new branch they need to get the entity base, dao base, service base, web.xml fixing and etc.

### CODE REVIEW

File: src/main/java/com/ciandt/techgallery/ofy/OfyService.java [hunk 1]
```diff
@@ -20,52 +21,54 @@
  */
 public class OfyService implements ServletContextListener {
 
-  /** Define all entities first. */
-  static {
+	/** Define all entities first. */
+	static {
+		ObjectifyService.register(Technology.class);
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/ofy/OfyService.java]: Don't use tabs for indenting.

File: src/main/java/com/ciandt/techgallery/persistence/dao/TechnologyDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.ciandt.techgallery.persistence.model.Technology;
+
+/**
+ * TechnologyDAOImpl methods implementation.
+ * 
+ * @author Felipe Goncalves de Castro
+ *
+ */
+public class TechnologyDAOImpl extends GenericDAOImpl<Technology, Long>
+		implements TechnologyDAO {
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/persistence/dao/TechnologyDAOImpl.java]: Don't use tabs for indenting.

File: src/main/java/com/ciandt/techgallery/persistence/model/BaseEntity.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import com.googlecode.objectify.annotation.Unindex;
+
+/**
+ * BaseEntity entity.
+ * 
+ * @author Felipe Goncalves de Castro
+ *
+ */
+public class BaseEntity<ID> {
+
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/persistence/model/BaseEntity.java]: Don't use tabs for indenting.

File: src/main/java/com/ciandt/techgallery/persistence/model/Technology.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Technology entity.
+ * 
+ * @author Felipe Goncalves de Castro
+ *
+ */
+@Entity
+public class Technology extends BaseEntity<Long> {
+
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/persistence/model/Technology.java]: Don't use tabs for indenting.

File: src/main/java/com/ciandt/techgallery/sample/service/TechServiceImpl.java [hunk 5]
```diff
@@ -32,21 +32,21 @@
    * @return key generated by datastore.
    */
   public TechResponse add(TechResponse entrada) {
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/sample/service/TechServiceImpl.java]: Please use English only for variables.

File: src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  /**
+   * POST for adding a technology.
+   */
+  @Override
+  public Response addTechnology(TechnologyResponse technology)
+      throws InternalServerErrorException, BadRequestException {
+    String techName = technology.getName();
+
+    if (techName == null || techName.equals("")) {
+      throw new BadRequestException("Nome da Tecnologia não pode ser em branco!");
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java]: Please use English only for messages.

File: src/main/webapp/WEB-INF/appengine-web.xml [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
-    <version>${app.version}</version>
-    <threadsafe>true</threadsafe>
-    <system-properties>
-        <property name="java.util.logging.config.file" value="WEB-INF/logging.properties"/>
-    </system-properties>
-    <static-files>
-        <exclude path="/node/**"/>
-        <exclude path="/node_modules/**"/>
-    </static-files>
+	<application>tech-gallery</application>
```
Comment (larruda) [inline src/main/webapp/WEB-INF/appengine-web.xml]: Switch from tab to spaces for indenting.

File: src/main/java/com/ciandt/techgallery/persistence/dao/TechnologyDAOImpl.java (hunk 2 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/persistence/dao/TechnologyDAOImpl.java]: Done

File: src/main/java/com/ciandt/techgallery/persistence/model/BaseEntity.java (hunk 3 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/persistence/model/BaseEntity.java]: Done

File: src/main/java/com/ciandt/techgallery/persistence/model/Technology.java (hunk 4 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/persistence/model/Technology.java]: Done

File: src/main/java/com/ciandt/techgallery/persistence/dao/GenericDAOImpl.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * GenericDAOImpl methods implementation.
+ * 
+ * @author Felipe Goncalves de Castro
+ *
+ */
+public class GenericDAOImpl<T extends BaseEntity<ID>, ID extends Serializable> implements
+    GenericDAO<T, ID> {
+  
+	private Class<T> clazz;
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/persistence/dao/GenericDAOImpl.java]: Don't use tabs for indenting.

File: src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  /**
+   * GET for getting all technologies.
+   */
+  @Override
+  public Response getTechnologies() throws InternalServerErrorException, NotFoundException {
+    List<Technology> techEntities = technologyDAO.findAll();
+    if (techEntities == null) {
+      throw new NotFoundException("Não foi encontrado nenhuma tecnologia!");
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java]: Please use English only for messages.

File: src/main/java/com/ciandt/techgallery/sample/service/TechServiceImpl.java (hunk 5 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/sample/service/TechServiceImpl.java]: Done

File: src/main/webapp/WEB-INF/appengine-web.xml (hunk 7 above)
Comment (felipegc) [inline src/main/webapp/WEB-INF/appengine-web.xml]: Done

File: src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java (hunk 6 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java]: Done

File: src/main/java/com/ciandt/techgallery/persistence/dao/GenericDAOImpl.java (hunk 8 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/persistence/dao/GenericDAOImpl.java]: Done

File: src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java (hunk 9 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/service/TechnologyServiceImpl.java]: Done

File: src/main/java/com/ciandt/techgallery/ofy/OfyService.java (hunk 1 above)
Comment (felipegc) [inline src/main/java/com/ciandt/techgallery/ofy/OfyService.java]: Done

**Pull request merged**